packages/e2e-test-utils-playwright/src/editor/insert-block.ts

```diff
@@ -40,7 +51,11 @@ async function insertBlock(
 		const block = recursiveCreateBlock( _blockRepresentation );
 
 		window.wp.data.dispatch( 'core/block-editor' ).insertBlock( block );
+
+		return block;
 	}, blockRepresentation );
+
+	return this.canvas.locator( `#block-${ clientId }` );
```

Comment:
I haven't tested this, but one issue I can see is that if you reload the page after block insertion and try to use this locator in assertion, it will fail.